### PR TITLE
PHPCS: php/datasource/class-fieldmanager-datasource-user.php

### PR DESCRIPTION
**Pull request by kjbenk**
# Issue
#271 

# Description
PHPCS for files:

- `php/datasource/class-fieldmanager-datasource-user.php`

### CODE REVIEW

File: php/datasource/class-fieldmanager-datasource-user.php [hunk 1]
```diff
@@ -114,146 +135,159 @@ public function get_value( $value ) {
...
+	 */
+	public function presave_alter_values( Fieldmanager_Field $field, $values, $current_values ) {
+		if ( 'post' != $field->data_type || ! $this->reciprocal || 'ID' != $this->store_property ) {
 			return $values;
 		}
 
 		if ( ! empty( $current_values ) ) {
 			foreach ( $current_values as $user_id ) {
-				delete_user_meta( $user_id, $this->reciprocal, $field->data_id );
+				delete_metadata( 'user', $user_id, $this->reciprocal, $field->data_id );
```
Comment (mboynes) [inline php/datasource/class-fieldmanager-datasource-user.php]: Why the change to `delete_metadata()` (and `add_metadata()` on line 240)?

Comment (kjbenk) [inline php/datasource/class-fieldmanager-datasource-user.php]: the linter complains and sends me here - https://vip.wordpress.com/documentation/vip/code-review-what-we-look-for/#working-with-wp_users-and-user_meta

https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/49cb5ce25f4712a2e3d2aa483736372c145f7921/WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php#L148-L158

Comment (mboynes) [inline php/datasource/class-fieldmanager-datasource-user.php]: that fools the linter, but doesn't fix the problem. we'll need to do something [like we did in the user context](https://github.com/alleyinteractive/wordpress-fieldmanager/blob/master/php/context/class-fieldmanager-context-user.php#L126-L148)
